src/test_array_builder.cc

```diff
@@ -310,61 +337,81 @@ Query::Status writeData(Context ctx, std::string array_name, tiledb_datatype_t d
     array = new Array(ctx, array_name, TILEDB_WRITE);
   else
     array = new Array(ctx, array_name, TILEDB_WRITE, encryption_type, encryption_key);
+
   Query query(ctx, *array, TILEDB_WRITE);
   query.set_layout(TILEDB_UNORDERED);
 
 
   // Hold a reference to the buffers so they exists until we are finished writting
   std::vector<std::pair<std::unique_ptr<std::vector<uint64_t>>, std::shared_ptr<void>>> buffers;
-
+  std::unique_ptr<std::vector<uint64_t>> offsets = std::unique_ptr<std::vector<uint64_t>>(new std::vector<uint64_t>);
```

Comment:
The `buffers` variable is a vector of pairs, where the first element in the pair is the offsets. Please re-use `buffers` rather than introducing this variable.

Trace the `addDataToQuery` routine for reference. It uses offsets and passes them back in a pair that is stored on `buffers`.